WordPress/WordPressShare/BlogPickerViewController.swift

```diff
@@ -0,0 +1,135 @@
+        navigationController?.popViewControllerAnimated(true)
+    }
+    
+    
+    // MARK: - Setup Helpers
+    private func setupView() {
+        title = NSLocalizedString("Site Picker", comment: "Title for the Site Picker")
+        preferredContentSize = UIScreen.mainScreen().bounds.size
+    }
+    
+    private func setupTableView() {
+        // Blur!
+        let blurEffect = UIBlurEffect(style: .Light)
+        tableView.backgroundColor = UIColor.clearColor()
+        tableView.backgroundView = UIVisualEffectView(effect: blurEffect)
+        tableView.separatorEffect = UIVibrancyEffect(forBlurEffect: blurEffect)
+        
+        // Fix: Hide the cellSeparators, when the table is empty
+        tableView.tableFooterView = UIView()
+    }
+    
+    private func setupNoResultsView() {
+        noResultsView = WPNoResultsView()
+        tableView.addSubview(noResultsView)
+    }
+    
+    
+    // MARK: - Private Helpers
+    private func loadSites() {
+        let authDetails = ShareExtensionService.retrieveShareExtensionConfiguration()
+        RequestRouter.bearerToken = authDetails!.oauth2Token as String
+        
+        let service = SiteService()
+        
+        showLoadingView()
+        
+        service.fetchSites { sites, error in
+            dispatch_async(dispatch_get_main_queue()) {
+                self.sites = sites
+                self.tableView.reloadData()
+                self.showEmptySitesIfNeeded()
+            }
+        }
+    }
+    
+    private func configureCell(cell: UITableViewCell, site: Site) {
+        // Site's Details
+        cell.textLabel?.text = site.name
+        cell.detailTextLabel?.text = site.URL.absoluteString.hostname()
+        
+        // Site's Blavatar
+        let placeholderImage = WPStyleGuide.Share.blavatarPlaceholderImage
+        
+        if let siteIconPath = site.icon, siteIconUrl = NSURL(string: siteIconPath) {
+            cell.imageView?.downloadBlavatar(siteIconUrl, placeholderImage: placeholderImage)
+        } else {
+            cell.imageView?.image = placeholderImage
+        }
+    }
+    
+    
+    // MARK: - No Results Helpers
+    private func showLoadingView() {
+        noResultsView.titleText = NSLocalizedString("Loading Sites...", comment: "Legend displayed when loading Sites")
+        noResultsView.hidden = false
+    }
+    
+    private func showEmptySitesIfNeeded() {
+        let hasSites = (sites?.isEmpty == false) ?? false
+        noResultsView.titleText = NSLocalizedString("No Sites", comment: "Legend displayed when the user has no sites")
+        noResultsView.hidden = hasSites
+    }
+    
+
+    // MARK: Typealiases
+    typealias PickerHandler = (siteId: Int, description: String?) -> Void
+    
+    // MARK: - Public Properties
+    var onChange                : PickerHandler?
+    
+    // MARK: - Private Properties
+    private var sites           : [Site]?
```

Comment:
Just a thought, what about `private var sites = [Site]()` to avoid the optional?